src/tests/ftest/osa/osa_online_parallel_test.py

```diff
@@ -0,0 +1,298 @@
+        """
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][3])
+
+    @fail_on(CommandFailure)
+    def get_pool_version(self):
+        """Get the pool version
+           Returns :
+            int : pool_version_value
+        """
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][4])
+
+    def daos_racer_thread(self, results):
+        """Start the daos_racer thread.
+        """
+        self.daos_racer = DaosRacerCommand(self.bin, self.hostlist_clients[0],
+                                           self.dmg_command)
+        self.daos_racer.get_params(self)
+        self.daos_racer.set_environment(
+            self.daos_racer.get_environment(self.server_managers[0]))
+        self.daos_racer.run()
+        results.put("Daos Racer Started")
+
+    def ior_thread(self, pool, oclass, api, test, flags, results):
+        """Start threads and wait until all threads are finished.
+        Args:
+            pool (object): pool handle
+            oclass (str): IOR object class
+            api (str): IOR api
+            test (list): IOR test sequence
+            flags (str): IOR flags
+            results (queue): queue for returning thread results
+
+        Returns:
+            None
+        """
+        processes = self.params.get("slots", "/run/ior/clientslots/*")
+        container_info = {}
+        mpio_util = MpioUtils()
+        if mpio_util.mpich_installed(self.hostlist_clients) is False:
+            self.fail("Exiting Test : Mpich not installed on :"
+                      " {}".format(self.hostfile_clients[0]))
+        self.pool = pool
+        # Define the arguments for the ior_runner_thread method
+        ior_cmd = IorCommand()
+        ior_cmd.get_params(self)
+        ior_cmd.set_daos_params(self.server_group, self.pool)
+        ior_cmd.dfs_oclass.update(oclass)
+        ior_cmd.api.update(api)
+        ior_cmd.transfer_size.update(test[2])
+        ior_cmd.block_size.update(test[3])
+        ior_cmd.flags.update(flags)
+
+        container_info["{}{}{}"
+                       .format(oclass,
+                               api,
+                               test[2])] = str(uuid.uuid4())
+
+        # Define the job manager for the IOR command
+        manager = Mpirun(ior_cmd, mpitype="mpich")
+        key = "".join([oclass, api, str(test[2])])
+        manager.job.dfs_cont.update(container_info[key])
+        env = ior_cmd.get_default_env(str(manager))
+        manager.assign_hosts(self.hostlist_clients, self.workdir, None)
+        manager.assign_processes(processes)
+        manager.assign_environment(env, True)
+
+        # run IOR Command
+        try:
+            manager.run()
+        except CommandFailure as _error:
+            results.put("FAIL")
+
+    def dmg_thread(self, puuid, rank, target, action, results):
```

Comment:
I see this method in osa_offline_parallel_test.py too. You might want to make it to a common method.

Reply:
I have ticket https://jira.hpdd.intel.com/browse/DAOS-5323?filter=-1 to move common methods to something like osa_utils.py later once I merge all the present scripts.

Reply:
Okay. Thanks.